Caltech-IPAC/firefly_client
# Firefly-1483: improve support for triview

#### [Firefly-1483](https://jira.ipac.caltech.edu/browse/FIREFLY-1483): change firefly_client to better support triview and plan to deprecate slate

see firefly PR: https://github.com/Caltech-IPAC/firefly/pull/1557
see jupyter_firefly_extension PR: https://github.com/Caltech-IPAC/jupyter_firefly_extensions/pull/29

  - firefly_client knows what sort of view it is using triview/slate/other
  - added change_triview_layout
  - warnings and noop when wrong function is used

#### Testing
   - https://fireflydev.ipac.caltech.edu/firefly-1483-triview/firefly
   - try `FireflyClient.make_client()` and `FireflyClient.make_lab_client()`
   - setting html_file parameter to `slate.html` will put it is slate mode otherwise it will be in triview mode
   - Using jupyter lab you can test with the following file in firefly_client/examples
      - make sure you change the url to: https://fireflydev.ipac.caltech.edu/firefly-1483-triview/firefly
      - `jl-lab-extension-demo-explicit2.ipynb`
      - `jl-lab-extension-slate-steps.ipynb`
      - `basic_3color.ipynb`
      - `basic-demo.ipynb`
      - `slate-demo-explicit.ipynb`
